src/sif/truckcost.cc

```diff
@@ -434,18 +434,20 @@ bool TruckCost::AllowedReverse(const baldr::DirectedEdge* edge,
 Cost TruckCost::EdgeCost(const baldr::DirectedEdge* edge,
                          const baldr::GraphTile* tile,
                          const uint32_t seconds) const {
-  auto speed = tile->GetSpeed(edge, flow_mask_, seconds);
+  auto edge_speed = tile->GetSpeed(edge, flow_mask_, seconds);
```

Comment:
this will automatically give you the truck speed if the edge has it

```suggestion
  auto edge_speed = tile->GetSpeed(edge, flow_mask_, seconds, true);
```